### PR TITLE
Zap blanket impl again

### PR DESCRIPTION
**Pull request by adetaylor**
Changes based on feedback [from PyO3](https://gist.github.com/davidhewitt/d0ed031fb05f6db98ee249ae089b268e)

### CODE REVIEW

File: text/0000-arbitrary-self-types-v2.md [hunk 1]
```diff
@@ -93,20 +114,7 @@ pub trait Receiver {
...
-    P: Deref<Target = T>,
-{
-    type Target = T;
-}
-```
-
-(See [alternatives](#no-blanket-implementation) for discussion of the tradeoffs here.)
-
-It is also implemented for `&T`, `&mut T`, `*const T` and `*mut T`.
+Implementations are provided for all standard library types which implement `Deref`. It is also implemented for `&T`, `&mut T`, `*const T` and `*mut T`.
```
Comment (Urhengulas) [inline text/0000-arbitrary-self-types-v2.md:117]: Do we have to do it for all libstd types that implement Deref? For the existing stable Self types (Pin, Box, ...) would be enough, wouldn't it?

File: text/0000-arbitrary-self-types-v2.md [hunk 2]
```diff
@@ -207,35 +215,27 @@ As this feature has been cooking since 2017, many alternative implementations ha
...
+## Add a blanket implementation for `Deref`
+[blanket-implementation]: #blanket-implementation
 
-The other major approach previously discussed is to have a `Receiver` trait, as proposed in this RFC, but without a blanket implementation for `T: Deref`. Possible advantages:
+This RFC fundamentally proposes allowing a `Receiver` trait so that some types can implement method dispatch even if they don't implement dereferencing.
 
-* It seems more in keeping with Rust norms where types can be in full control of the traits they implement
-* It would allow a type to decide that it wants to be dereferencable, without allowing method calls on the inner type
-* It would allow a type to specify a different `Target` for `Deref` vs `Receiver`.
+This permits four permutations for a type `A``:
```
Comment (Urhengulas) [inline text/0000-arbitrary-self-types-v2.md:225]:
```suggestion
This permits five permutations for a type `A``:
```

File: text/0000-arbitrary-self-types-v2.md [hunk 3]
```diff
@@ -207,35 +215,27 @@ As this feature has been cooking since 2017, many alternative implementations ha
...
 
-It's easy to imagine cases where this would be confusing for users. Imagine a type `A` dereferences to `B` yet allows method calls only on `C` (or perhaps more confusingly, allows method calls only on `Pin<&mut B>`).
+Nevertheless, there are known use-cases for case 5, so this flexibility is required. This fifth case is required when a type doesn't really "contain" another type, but instead is used to represent inheritance hierarchies or other types of relationship. For instance, in [PyO3](https://github.com/PyO3/pyo3), the `Py` smart pointer [may "contain" more specific Python types](https://gist.github.com/davidhewitt/d0ed031fb05f6db98ee249ae089b268e) such as `Py<PyList>`. In this case, we'd like `Deref::Target` to point to a more general Python type, `Py<PyAny>`, while we'd still like to be able to call `PyList` methods directly on `Py` - all the while without creating a reference to an underlying `PyAny` or `PyList`.
```
Comment (Urhengulas) [inline text/0000-arbitrary-self-types-v2.md:234]: `Deref::Target` or `Receiver::Target`?

**Comment by adetaylor** [comment]
This is now old and obsolete...

**Pull request closed without merging**